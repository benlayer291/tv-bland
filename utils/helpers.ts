export const removeProtocolFromUrl = (url: string): string =>
  url.replace(/^https?:\/\//i, '//')

export const removeHtmlTagsFromString = (htmlString: string): string =>
  htmlString.replace(/(<([^>]+)>)/gi, '')

export const checkDeviceIsMobile = (): boolean => {
  return (
    typeof navigator !== 'undefined' &&
    /Android|Mobi/i.test(navigator.userAgent)
  )
}
