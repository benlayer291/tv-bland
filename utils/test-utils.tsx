import { render, RenderOptions } from '@testing-library/react'

type Props = {
  children: React.ReactElement
}

const Providers: React.FunctionComponent<Props> = ({ children }: Props) => {
  return <>{children}</>
}

const customRender = (
  ui: React.ReactElement,
  options: Omit<RenderOptions, 'queries'> = {}
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
