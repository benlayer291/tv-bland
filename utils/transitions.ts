type Transtion = {
  duration: number
  ease: number[]
}

export const transition: Transtion = {
  duration: 5,
  ease: [0.43, 0.13, 0.23, 0.96],
}
