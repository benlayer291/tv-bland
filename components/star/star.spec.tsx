import React from 'react'
import { render, screen } from '../../utils/test-utils'

import Star from './star'

describe('<Star />', () => {
  it('renders correctly inactive', () => {
    render(<Star />)

    expect(screen.getByTestId('star')).toBeInTheDocument()
  })

  it('renders correctly active', () => {
    render(<Star isActive={true} />)

    expect(screen.getByTestId('star')).toBeInTheDocument()
    expect(screen.getByTestId('star')).toHaveClass('Star--is-active')
  })

  it('matches snapshot', () => {
    render(<Star />)

    expect(screen.getByTestId('star')).toMatchSnapshot()
  })
})
