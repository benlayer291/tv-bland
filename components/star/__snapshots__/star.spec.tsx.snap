// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Star /> matches snapshot 1`] = `
<svg
  class="Star"
  data-testid="star"
  viewBox="0 0 14 12"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clip-rule="evenodd"
    d="M6.667 9.95L2.547 12l.786-4.342L0 4.584l4.607-.634L6.667 0l2.06 3.95 4.606.634L10 7.658 10.787 12l-4.12-2.05z"
    fill-rule="evenodd"
  />
</svg>
`;
